### PR TITLE
Migrate to Jest

### LINKED ISSUE #24: Migrate to Jest

### PR DESCRIPTION
**Pull request by sibiraj-s**
Jest

### Changes done.

- switch to jest (test run in both node and jsdom environments)
- remove grunt. 

Closes #24

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -63,7 +63,9 @@
   ],
   "scripts": {
     "build": "rollup -c",
-    "test": "npm run build && grunt test",
+    "test-node": "jest --verbose --environment=node",
```
Comment (DanielRuf) [inline package.json:66]: Hm, not sure here. Personally I have a favor to use `test:node` and `test:web`.

Comment (sibiraj-s) [reply package.json]: Changed.

**Comment by sibiraj-s** [comment]
@DanielRuf . Can I have permissions for branch protection settings? or can you enable the following.

- require pull request to pass the tests
- require at-least one review to merge the PR?

File: package.json (hunk 1 above)
Comment (sibiraj-s) [reply package.json]: Updated

**Comment by DanielRuf** [comment]
> @DanielRuf . Can I have permissions for branch protection settings? or can you enable the following.
> 
>     * require pull request to pass the tests
> 
>     * require at-least one review to merge the PR?

You are free to make these changes as needed. It's ok from my side.

**Comment by sibiraj-s** [comment]
> > @DanielRuf . Can I have permissions for branch protection settings? or can you enable the following.
> > ```
> > * require pull request to pass the tests
> > 
> > * require at-least one review to merge the PR?
> > ```
> 
> You are free to make these changes as needed. It's ok from my side.

Thanks.

**Comment by sibiraj-s** [comment]
Can you leave your approval if the changes are fine. I will merge it. No more changes in this PR.

**Review by DanielRuf — APPROVED** [review]

**Pull request merged**